crates/services/txpool_v2/src/storage/mod.rs

```diff
@@ -36,34 +38,53 @@ pub struct StorageData {
 
 pub type RemovedTransactions = Vec<PoolTransaction>;
 
+pub trait Collision<StorageIndex> {
```

Comment:
Linked with the two others comments I left on this file : 

If we remove this type in parameter of function his has no reasons to exists here anymore + in any case for me the storage shouldn't have the definition of a collision directly defined on his sub-module.

Reply:
https://github.com/FuelLabs/fuel-core/pull/2162#discussion_r1779923513

The main problem is the `can_store_transaction` method also verifies the collision. If we don't want to leak the existence of the collision to the storage, then we should do this check on the pool level.

Moved this check in https://github.com/FuelLabs/fuel-core/pull/2162/commits/47ebb4cbd5975e01f84a58af733aacdf38215d71 to TxPool level. Now it looks cleaner.